dor/adapters/bag_adapter.py

```diff
@@ -16,27 +18,24 @@ def __init__(self, message: str):
         self.message = message
 
 
-class FakeBagAdapter():
-
-    def __init__(self, identifier: str) -> None:
-        self.identifier = identifier
-
-    def validate(self) -> None:
-        return
-    
-    @property
-    def dor_info(self) -> dict:
-        return {
-            'Root-Identifier': "00000000-0000-0000-0000-000000000001"
-        }
-
-
 class BagAdapter:
     
     dor_info_file_name = "dor-info.txt"
 
-    def __init__(self, path: Path, file_provider: FileProvider) -> None:
-        self.bag = bagit.Bag(str(path))
+    @classmethod
+    def load(cls, path: Path, file_provider: FileProvider) -> Self:
+        bagit_bag = bagit.Bag(str(path))
+        return cls(bagit_bag, file_provider)
+
+    @classmethod
+    def make(cls, payload_path: Path, dor_info: dict[str, str], file_provider: FileProvider) -> Self:
+        bagit_bag = bagit.make_bag(str(payload_path))
+        bagit._make_tag_file(payload_path / cls.dor_info_file_name, dor_info)
+        bagit_bag.save()
+        return cls(bagit_bag, file_provider)
```

Comment:
I wasn't sure if this method should also create `dor-info.txt`, but I guess I figured we'd want to create each time we make bags and `validate` requires the file. Interested in thoughts on this.

Reply:
I could see using it in the sample builder.
The only scenario I can see where we might generate bags without `dor-info.txt` is replication, in which case the `dor_info` parameter could be optional.

Reply:
So as of now, if you don't create `dor-info.txt`, the bag will fail validation. So we'd have to relax that if we wanted to use this in other contexts.

Reply:
How about something like this? It probably does make sense to have a separate method and allow bags without `dor-info.txt` to be valid. https://github.com/mlibrary/dor-py/pull/40/commits/f18b8903e89aeb0b03bc85bcb1c23cc298b7d09b

Reply:
I just updated the test cases a bit. Was there something you wanted me to do before this is approved? I would like to start using this in the `Packager` code (note this is against the `generate-packages` branch).